int64 serial_id
location goal
---
location[] path
